### DISCUSSION TITLE
2007/04/02/%E6%8A%80%E6%9C%AF%E6%96%87%E7%AB%A0%E5%92%8C%E8%AE%A8%E8%AE%BA

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#148] [Announcements]
# 2007/04/02/%E6%8A%80%E6%9C%AF%E6%96%87%E7%AB%A0%E5%92%8C%E8%AE%A8%E8%AE%BA

新的页面地址：http://ruanqizhen.wordpress.com/technology/

https://blog.qizhen.xyz/2007/04/02/%E6%8A%80%E6%9C%AF%E6%96%87%E7%AB%A0%E5%92%8C%E8%AE%A8%E8%AE%BA

**Comment by ruanqizhen** [comment]
之前博客下的留言：


Qizhen
2010年06月2日 16:16 编辑
http://ruanqizhen.spaces.live.com/blog/cns!5852D4F797C53FB6!4743.entry

回复
Michael
2010年06月2日 15:16 编辑
阮老师您好，我有一本您的“我和LabView” ，在 "4.8.5 借助 C 语言" 这节中提到 ： “ ……LabView传递的只是这数据空间地址值。后续程序若需要用到被引用数据，则可以通过地址值 得到数据的内容 ………… 见图 4.65”， 对图4.65我没看明白到底是如何通过地址取到数据的，我目前开发的一个程序就需要这样的功能，LavView 调用DLL(c++) 返回一个c分配的内存的地址，而后在LabView中通过这个地址取到值。

回复
Qizhen
2010年06月2日 11:26 编辑
你是使用软件来定时采集数据吗？软件定时是非常不精确的，这个问题没法从根本上解决，调整线程优先级也不会有太大改善。不过，使用定时循环结构来代替延时函数，精度会有所提高。

回复
peng
2010年06月1日 09:42 编辑
阮老师您好，前些天购买了您出的《我和LabVIEW》这本书，现在有个问题想请教下您！我的实验涉及用数据采集卡采集信号，而且我想在对压电陶瓷进行单次调谐的同时进行数据采集，使两者时间上达到匹配，但是我发现每次采集时间都不同。我实现采集的那段程序是：（1）采集卡输出口给压电陶瓷一个高电平，（2）调用库函数进行数据采集，这块有一个时间为采集时间的延时函数，（3）再用采集卡输出口给压电陶瓷一个低电平。这块我用的是顺序结构。我咨询过Labview的工程师，他说由于在程序运行的时候会出现多线程问题，当其他线程也在运行的时候就会使数据采集卡的采集时间缩短，如果在采集的时候其他线程没有被执行这样采集时间就不会受影响。我这个用Labview搭建的实验平台只有一个主VI，有几个子VI。我看到您书上写的可以设置VI优先级，但我这个只有一个主VI,是不是一个主VI是一个线程啊，这样的话是不是就不涉及什么线程之间抢占资源的问题了呢？很困惑啊。。。

回复
Qizhen
2010年05月24日 11:25 编辑
Daisy Dai，你的问题我还不是很清楚。如果只是找出相同元素，归类，那么LabVIEW没有一个现成的函数，一步完成。自己编程，比较数组中的每个元素，然后按照值，copy到不同的分类数组中去就行了。

回复
Daisy
2010年05月20日 13:57 编辑
阮老师，您好，想请教一下使用LabVIEW 7怎样对一组数据例：[1，1，2，1，3，2，4，1，1，2，3，4]进行分类成：a1[1，1，1，1]，b1[2,2,2]，c1[3，3]，d1[4，4]，和4种数e1[1,2,3,4]。此问题困扰我很久了，非常感谢！

回复
Qizhen
2010年04月22日 12:47 编辑
我不太了解有关模糊神经网络工具包方面的内容，不确定LabVIEW是否有类似的功能。这需要去询问NI的销售人员。

回复
娜
2010年04月17日 21:11 编辑
您好，请问一下，labview中有模糊神经网络工具包吗？它能实现在MATLAB中调用ANFIS的fi文件吗？怎样调用MATLAB中的fis文件

回复
Qizhen
2009年10月9日 11:11 编辑
厂家是否修复取决于厂家的产品战略。但是，像这种驱动程序的问题，终端客户是无法自己解决的。如果厂家不修复，客户只能自己尝试一下别的编程方法，看是否能把问题绕开。

回复
Unknown
2009年10月1日 12:13 编辑
谢谢阮老师的回答，不是每次采集都有这样的问题，就是重复采集一定次数后，会有这样的问题。不过最近采集数据时，labview经常会自动关闭，dll不是我们写的，是厂家给提供的。他们会给修正吗？可能周期会很长吧。

回复
Qizhen
2009年09月29日 17:04 编辑
如果有时运行时正确的，问题只有在重复一定次数的情况下才能出现。最有可能是DLL函数有bug，运行过程中崩溃。这个问题调试和解决起来可能比较麻烦，如果DLL不是你们写的，应当把问题反馈给生产DLL的厂家，让他们进行修正。

回复
Unknown
2009年09月29日 14:03 编辑
阮老师： 你好，对你的书期待已久，明天我就去书店拜读一下您的巨作，我是西安的一名学生，目前正在做一个虚拟仪器项目，我现在遇到一个问题，在调用数据采集卡动态函数库时，经常在采集数据时，LabVIEW软件就会自动关闭，请问这可能是什么原因啊？我检查过打开设备和关闭设备都没有问题。请阮老师根据多年从业经验给提一下指导性的建议。

回复
Qizhen
2009年08月27日 10:41 编辑
所有软件安装的时候都会写注册表信息的。只删文件是没有用的。正确卸载方法是在控制面板的增加和删除程序中找到 “National Instruments 软件”一项，所有NI软件都在这里管理。但是，由于你已经采用错误方法删除了相关文件，现在这个方法可能已经不起作用了。你只好手工打开注册表，搜索 “National Instruments“，”NI“，”LabVIEW“等关键字，然后把它们都删干净。

回复
Zhaogang
2009年08月21日 20:23 编辑
各位前辈，我今天上午装了Labview 8.6。在装的时候我发现C盘空间太小，就装在了D盘。随后我发现D盘太乱，就想删除重装。谁知道就是找不到哪里可以卸载这个软件。我从开始/programs 中，以及设置中的增加和删除程序中都找过，就是没有发现那里可以直接卸载，后来我干脆直接把D盘中的程序删除了。可是我再想重新安装Labview 就是无法再安装了。这到底是怎么回事呢？我后来把D格式化了，然后有用超级兔子等尝试删除了多余的注册表，但还是安装不上这个软件。这到底是为什么呢？就差重新安装系统了。请各位帮帮我。衷心感谢了！sherryvictor

回复
Jun
2008年11月22日 11:07 编辑
阮大哥你好！想请教一个问题，我的程序中用到工具栏，在事件结构中响应工具栏中按钮的事件。后来我又想添加菜单栏，其中有一些菜单和工具栏中按钮功能相同，请问一下如何才能不另外写菜单栏的事件响应程序，直接把它整合到按钮的响应事件中去呢？谢谢！

回复
Qizhen
2008年01月19日 12:38 编辑
我平时接触的客户不多，所以没有这方面的案例。但是你说的这种应用，用 LabVIEW 实现是肯定没有问题的。你可以和 NI 的销售联系，他们会有案例的。

回复
xk
2008年01月17日 10:20 编辑
    想请教一下LabView能开发实时在线的监测系统吗？
    我们最近想开发一套振动的在线监测系统，实现长时间的在线监测，稳定运行一个月以上。数据量大概每分钟1M的数据。需要数据库的存储功能，网络上的实时波形的传输功能。现在有没有这方面的案例？
    谢谢！

回复
